man/systemd.exec.xml

```diff
@@ -1234,13 +1234,15 @@
       <varlistentry>
         <term><varname>NoNewPrivileges=</varname></term>
 
-        <listitem><para>Takes a boolean argument. If true, ensures
-        that the service process and all its children can never gain
-        new privileges. This option is more powerful than the
-        respective secure bits flags (see above), as it also prohibits
-        UID changes of any kind. This is the simplest, most effective
-        way to ensure that a process and its children can never
-        elevate privileges again.</para></listitem>
+        <listitem><para>Takes a boolean argument. If true, ensures that the service
+        process and all its children can never gain new privileges. This option is more
+        powerful than the respective secure bits flags (see above), as it also prohibits
+        UID changes of any kind. This is the simplest most effective way to ensure that a
+        process and its children can never elevate privileges again. Defaults to true in
+        the user manager instance if <varname>SystemCallFilter=</varname>,
+        <varname>SystemCallArchitectures=</varname>, or
+        <varname>RestrictAddressFamilies=</varname> are specified, and false otherwise.
+        </para></listitem>
```

Comment:
Hmm, NNP doesn't just "default" to "yes" if SystemCallFilter= is set in --user mode. It actually is entirely overriden in that case. If the user combines SystemCallFilter= in user mode with NNP=no, then the latter will have no effect and result will be NNP=yes after all.

Also, ProtectKernelTunables=, ProtectKernelTunables=, MemoryDenyWriteExecute=, PrivateDEvices=, RestrictRealtime also force NNP=yes.